### DISCUSSION TITLE
list widget

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#815] [Ideas]
I searched all the widgets and found that none of them fit the list style, can you implement a list widget? I've thought about laying out a bunch of buttons vertically, and then whether I can style it to debug the distance between each two buttons I don't know

![{109F8700-3D33-42be-A1F5-32FCD415450C}](https://github.com/goki/goki/assets/19886504/50e34a07-af82-4d2f-8c71-91541c905ff3)
![{BB28CEB6-AF01-4380-9CA2-5804D703EFAB}](https://github.com/goki/goki/assets/19886504/d00f9138-fbb1-43e5-9b35-225050c46cf9)

**Comment by ddkwork** [comment]
![{0E5F8C7F-E673-434e-87F4-F7EA10187CDA}](https://github.com/goki/goki/assets/19886504/af5d1102-e804-4cc1-86e1-9e2c04b28a22)
For the layout display of the registers, I have thought about splitting the layout horizontally, and the left side is displayed with a struct, but I found that each field of the struct should correspond to the view on the right, so can I add an extension property to the struct display widget to display the extended information of each field, and use this idea to lay out the registers, please give me some instructions, thank you

**Comment by ddkwork** [comment]
I've thought about using custom formatting fields to do this, but when we change the value of a register, the content in the right layout should be dynamically fetched and populated based on whether the register is a pointer or not, so the custom formatting method is still not suitable for this situation

**Comment by kkoreilly** [comment]
If the information is organized such that each row follows the same pattern and has columns that can be aligned with those of other rows, you should use a table view. If not, you could use a slice view on a slice of strings that contains all of the data for each row. Also, you could combine all of the data into one formatted string that is displayed in a text editor, which is what it looks like your reference app is doing. I can give you more information about how to do any of these options if you need.

**Comment by ddkwork** [comment]
> If the information is organized such that each row follows the same pattern and has columns that can be aligned with those of other rows, you should use a table view. If not, you could use a slice view on a slice of strings that contains all of the data for each row. Also, you could combine all of the data into one formatted string that is displayed in a text editor, which is what it looks like your reference app is doing. I can give you more information about how to do any of these options if you need.

I'm sorry for the late reply, yesterday I had something to do and didn't have time to go online all day, I'll test it as soon as possible according to your prompts

**Comment by ddkwork** [comment]
I found out today that the task manager and file manager, as well as the etable and svg visual editor-related repositories, seem to have been removed, and these repositories are still very common programs

**Reply by kkoreilly** [reply]
We are currently renaming the framework from Goki to Cogent Core, so some things are currently not where they will end up. We are not removing the repositories you mentioned, so please do not be worried. The task manager, file manager, and svg editor will end up in https://github.com/cogentcore/cogent in the next couple of hours. Etable is currently at https://github.com/emer/etable and will also probably be moved to cogentcore soon. We are not removing any apps or functionality, and everything should be stable soon. You should update all of your import paths in your apps by doing a find-and-replace from `goki.dev` to `cogentcore.org/core`. Let me know if you have any other questions or problems.

**Reply by ddkwork** [reply]
> We are currently renaming the framework from Goki to Cogent Core, so some things are currently not where they will end up. We are not removing the repositories you mentioned, so please do not be worried. The task manager, file manager, and svg editor will end up in https://githubfast.com/cogentcore/cogent in the next couple of hours. Etable is currently at https://githubfast.com/emer/etable and will also probably be moved to cogentcore soon. We are not removing any apps or functionality, and everything should be stable soon. You should update all of your import paths in your apps by doing a find-and-replace from `goki.dev` to `cogentcore.org/core`. Let me know if you have any other questions or problems.

![{00106590-830F-43b9-BFA5-B3D3EF3FBC9F}](https://github.com/cogentcore/core/assets/19886504/dc8b931f-eb61-4c1a-9078-643a544fe369)
Module import failed

**Reply by kkoreilly** [reply]
You need to do `go get cogentcore.org/core@main`

**Reply by kkoreilly** [reply]
Also, you should install the `core` tool and use it instead of the `goki` tool (`go install cogentcore.org/core/core@main`).

**Reply by ddkwork** [reply]
thx you

**Comment by ddkwork** [comment]
At present, goki has tested the complex layout of multiple scenes on my side, I will take the time to learn goki's drawing function, once a little received, I will talk about collaborating with you to design tree table widgets and other things, I use table controls for cryptography algorithm testing applets, packet capture programs, debugger programs, and may also drag and drop visual interface designers in the future, as well as some ERP software, such as warehouse management, purchase, sales and inventory, etc., There are also some excel editors and so on, for excel editors, I personally don't think WPS, Microsoft's excel editor, etc. are more modern design, because their excel editor needs to manually enter the sum of sums, for classification it is a mess, there is no pace to keep up with the times, on the contrary, suppose we use the tree table widget for the design of ERP software, excel editor, financial statement related software is quite intuitive, For example, the excel editor uses tree table widgets to automatically classify, sum, and many more intuitive features. When I get to grips with the drawing logic, I think I'll first study the table row click event support and odd and even row zebra crossings, select the row and set the background color, but unfortunately I'm still in the testing layout stage, but I'll try to learn your code, because I've studied about seven or eight UI libraries before, and Goki is the most in line with the taste of the Go language, haha.

**Reply by kkoreilly** [reply]
Sounds good! I would be glad to work with you on expanding the capabilities of the framework, and I think all of those features sound good.

**Comment by ddkwork** [comment]
> If the information is organized such that each row follows the same pattern and has columns that can be aligned with those of other rows, you should use a table view. If not, you could use a slice view on a slice of strings that contains all of the data for each row. Also, you could combine all of the data into one formatted string that is displayed in a text editor, which is what it looks like your reference app is doing. I can give you more information about how to do any of these options if you need.

I will try, thx  you very much

**Comment by ddkwork** [comment]
> If the information is organized such that each row follows the same pattern and has columns that can be aligned with those of other rows, you should use a table view. If not, you could use a slice view on a slice of strings that contains all of the data for each row. Also, you could combine all of the data into one formatted string that is displayed in a text editor, which is what it looks like your reference app is doing. I can give you more information about how to do any of these options if you need.

Based on your prompts, I thought for a long time and decided to use a table layout for the register layout, because if you use a text editor, it will not reflect the user's behavior of changing the register value. Another layout is that the left struct shows the widget display register, and the right text editing widget, so it is possible to insert the split widget in the process of adjusting the layout, and the register layout should not have a separator for the split widget because the position is tight, so the text editor method is not used. I'll try to lay out the register area first, and I'll ask you about the list widget when the time comes
